server/public/model/config.go

```diff
@@ -382,6 +382,7 @@ type ServiceSettings struct {
 	EnableAPITeamDeletion                             *bool
 	EnableAPITriggerAdminNotifications                *bool
 	EnableAPIUserDeletion                             *bool
+	EnableAPIPostDeletion                             *bool
```

Comment:
It's fine for consistency, but I still wonder why we need these settings at all. Should they be permission settings?

Reply:
I think it makes sense to have these as configs and separate from the permissions. With the permanent nature of these deletions I think the extra safe guard makes sense